### PR TITLE
Update v1.3 feature about new command owe

### PR DESCRIPTION
**Pull request by LifHoshi**
Fixes https://github.com/AY2324S2-CS2103T-W11-2/tp/issues/102.

The new command owe is the command that checks the amount of money each person should submit. 

The format of the new command is: owe INDEX m/MONEY e.g. owe 3 m/20.

The Money part can maximum support 2 decimal digits.

**Original graph: **
image

**After command: **
image

**Possible invalid commands handle: **
image
image
image

### CODE REVIEW

**Comment by LifHoshi** [comment]
Check fail due to some UG format

**Comment by Fidget-Spinner** [comment]
Can you open an issue and link it to that?

**Comment by LifHoshi** [comment]
> Can you open an issue and link it to that?

Okay done

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 1]
```diff
@@ -60,10 +61,10 @@ public void parse_allFieldsPresent_success() {
 
         // multiple roles - all accepted
         Person expectedPersonMultipleRoles = new PersonBuilder(BOB).withRoles(VALID_ROLE_FRIEND, VALID_ROLE_HUSBAND)
-                .build();
+                .withAmount("0.0").build();
         assertParseSuccess(parser,
                 NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB + ROLE_DESC_HUSBAND
-                        + ROLE_DESC_FRIEND + CCA_DESC_DEFAULT,
+                        + ROLE_DESC_FRIEND + CCA_DESC_DEFAULT,
```
Comment (Fidget-Spinner) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:67]:
```suggestion
                        + ROLE_DESC_FRIEND + CCA_DESC_DEFAULT,
```

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 2]
```diff
@@ -27,22 +28,27 @@ public class TypicalPersons {
...
     public static final Person CARL = new PersonBuilder().withName("Carl Kurz").withPhone("95352563")
             .withEmail("heinz@example.com").withAddress("wall street")
-            .withCcas("NUS Cycling").build();
+            .withCcas("NUS Cycling")
+            .withAmount("30.00").build();
     public static final Person DANIEL = new PersonBuilder().withName("Daniel Meier").withPhone("87652533")
             .withEmail("cornelia@example.com").withAddress("10th street").withRoles("friends")
-            .withCcas("NUS Cycling").build();
+            .withCcas("NUS Cycling")
+            .withAmount("40.00").build();
```
Comment (Fidget-Spinner) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:46]:
```suggestion
            .withAmount("40.00").build();
```

**Review by Fidget-Spinner — COMMENTED** [review]
Tests are failing because of trailing whitespace below. See both comments below.

File: src/main/java/seedu/address/logic/commands/OweCommand.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
+package seedu.address.logic.commands;
```
Comment (Fidget-Spinner) [inline src/main/java/seedu/address/logic/commands/OweCommand.java:1]: Can you add a comment at the top of the file crediting which file you copied from? Also credit CS2103T teaching team (if you copied from a file from the original AB3)

File: src/main/java/seedu/address/logic/parser/OweCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package seedu.address.logic.parser;
```
Comment (Fidget-Spinner) [inline src/main/java/seedu/address/logic/parser/OweCommandParser.java:1]: Same here needs credit (only if you copied)

File: src/main/java/seedu/address/storage/JsonAdaptedAmount.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package seedu.address.storage;
```
Comment (Fidget-Spinner) [inline src/main/java/seedu/address/storage/JsonAdaptedAmount.java:1]: Same needs credits (only if copied).

File: src/test/java/seedu/address/logic/commands/OweCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.commands;
```
Comment (Fidget-Spinner) [inline src/test/java/seedu/address/logic/commands/OweCommandTest.java:1]: Needs credits.

File: src/test/java/seedu/address/logic/parser/OweCommandParserTest.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
+package seedu.address.logic.parser;
```
Comment (Fidget-Spinner) [inline src/test/java/seedu/address/logic/parser/OweCommandParserTest.java:1]: Same here needs cddredits if copied.

File: src/test/java/seedu/address/model/amount/AmountTest.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.amount;
```
Comment (Fidget-Spinner) [inline src/test/java/seedu/address/model/amount/AmountTest.java:1]: Ditto.

**Review by Fidget-Spinner — APPROVED** [review]

**Pull request merged**